Lib/test/test_msvcrt.py

```diff
@@ -61,34 +61,48 @@ def test_get_osfhandle(self):
 
 
 class TestConsoleIO(unittest.TestCase):
+    def run_in_separated_process(self, code):
+        # Run test in a seprated process to avoid stdin conflicts.
+        # See: gh-110147
+        cmd = [sys.executable, '-c', code]
+        subprocess.run(cmd, check=True, capture_output=True,
+                       creationflags=subprocess.CREATE_NEW_CONSOLE)
+
+
     def test_kbhit(self):
-        h = msvcrt.get_osfhandle(sys.stdin.fileno())
-        flush_console_input_buffer(h)
-        self.assertEqual(msvcrt.kbhit(), 0)
+        code = dedent('''
+            import msvcrt
+            assert msvcrt.kbhit() == 0
+        ''')
+        self.run_in_separated_process(code)
 
     def test_getch(self):
         msvcrt.ungetch(b'c')
         self.assertEqual(msvcrt.getch(), b'c')
 
     def test_getwch(self):
```

Comment:
test_getwche() looks like a copy/paste, you just replaced getwch() with getwche(). Can you rename the function to check_getwch(func), so test_getwch() would call check_getwch("getwch") and test_getwch() would call check_getwch("getwche")?